var/spack/repos/builtin/packages/py-cython/package.py

```diff
@@ -45,6 +48,9 @@ class PyCython(PythonPackage):
     version("0.23.5", sha256="0ae5a5451a190e03ee36922c4189ca2c88d1df40a89b4f224bc842d388a0d1b6")
     version("0.23.4", sha256="fec42fecee35d6cc02887f1eef4e4952c97402ed2800bfe41bbd9ed1a0730d8e")
 
+    # cython -M uses distutils, which is removed in Python 3.12
+    depends_on("python@:3.11", when="@:3.0.2", type=("build", "link", "run"))
```

Comment:
I still see a ton of distutils imports in 3.0.3...

Reply:
You're right: https://github.com/cython/cython/issues/5751

Reply:
Should we add an unconditional upper bound on 3.11 for now?

Reply:
Yeah, better. Pushed that.